### PR TITLE
Validate SDL directives

### LINKED ISSUE #1389: add validation of directives in SDL
From #1383:

> It's very critical to add validation of directives in SDL since as [noted here](https://github.com/facebook/graphql/issues/429#issuecomment-392990065) no one validates them correctly at the moment. That will lead to a situation very similar to the recent `Allow interfaces to have no implementors`. Since spec defines set of validation rules but no one implements them. Probably we shouldn't enforce `Directives Are Unique Per Location` until https://github.com/facebook/graphql/issues/429 resolved but all other validation should be implemented e.g. require the schema to contain applied directives, correct directive arguments, etc.

### PR DESCRIPTION
**Pull request by mjmahone**
Re-uses the existing `validateSchema` to resolve #1389.

This is a sub-optimal solution: ideally we'd be doing validation purely on the AST, as @IvanGoncharov is working on. However, this gets us to a state where people aren't using directives illegally within the SDL, which should allow us to release them.

### CODE REVIEW

File: src/type/validate.js [hunk 1]
```diff
@@ -165,7 +179,10 @@ function getOperationTypeNode(
   return type.astNode;
 }
 
-function validateDirectives(context: SchemaValidationContext): void {
+function validateDirectiveDefinitions(context: SchemaValidationContext): void {
+  // Ensure no directive is defined multiple times
+  const directiveDefinitions = new Map();
+
```
Comment (IvanGoncharov) [inline src/type/validate.js]: @mjmahone This change is make sense on it's own since you can manually construct `GraphQLSchema` with duplicated directives.
Can you extract this change into separate PR?

Comment (IvanGoncharov) [reply src/type/validate.js]: On `Map`: all around this library, we use `Object.create(null)` for maps with strings as keys and `Map` for non-string keys. AFAIK `Map` have significantly lover performance, I'm also not sure about memory usage.

**Comment by mjmahone** [comment]
#1438 is strictly better than this PR, so closing
